src/transfer.cpp

```diff
@@ -544,51 +542,53 @@ void Transfer::complete()
             }
         }
 
-        if (!fixedfingerprint && success && fa->fopen(&localfilename, true, false))
+        if (auto fa = client->fsaccess->newfileaccess())
         {
-            fingerprint.genfingerprint(fa);
-            if (isvalid && !(fingerprint == *(FileFingerprint*)this))
+            if (!fixedfingerprint && success && fa->fopen(&localfilename, true, false))
             {
-                LOG_err << "Fingerprint mismatch";
-
-                // enforce the verification of the fingerprint for sync transfers only
-                if (syncxfer && (!badfp.isvalid || !(badfp == fingerprint)))
+                fingerprint.genfingerprint(fa.get());
+                if (isvalid && !(fingerprint == *(FileFingerprint*)this))
                 {
-                    badfp = fingerprint;
-                    delete fa;
-                    chunkmacs.clear();
-                    client->fsaccess->unlinklocal(&localfilename);
-                    return failed(API_EWRITE);
-                }
-                else
-                {
-                    // We consider that mtime is different if the difference is >2
-                    // due to the resolution of mtime in some filesystems (like FAT).
-                    // This check prevents changes in the fingerprint due to silent
-                    // errors in setmtimelocal (returning success but not setting the
-                    // modification time) that seem to happen in some Android devices.
-                    if (abs(mtime - fingerprint.mtime) <= 2)
+                    LOG_err << "Fingerprint mismatch";
+
+                    // enforce the verification of the fingerprint for sync transfers only
+                    if (syncxfer && (!badfp.isvalid || !(badfp == fingerprint)))
                     {
-                        fixfingerprint = true;
+                        badfp = fingerprint;
+                        chunkmacs.clear();
+                        client->fsaccess->unlinklocal(&localfilename);
+                        fa.reset();
```

Comment:
Let's move this back to before `chunkmacs.clear()`?

Reply:
good catch! done